packages/devtools_app/lib/src/screens/memory/framework/memory_screen.dart

```diff
@@ -38,16 +40,27 @@ class MemoryBody extends StatefulWidget {
   MemoryBodyState createState() => MemoryBodyState();
 }
 
-class MemoryBodyState extends State<MemoryBody> {
+class MemoryBodyState extends State<MemoryBody>
+    with ProvidedControllerMixin<MemoryController, MemoryBody> {
   @override
   void initState() {
     super.initState();
     ga.screen(MemoryScreen.id);
   }
 
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    initController();
+  }
+
   @override
   Widget build(BuildContext context) {
-    // TODO(polina-c): load static body if not connected.
-    return const ConnectedMemoryBody();
+    if (controller.isOffline ||
+        !serviceConnection.serviceManager.hasConnection) {
+      return const OfflineMemoryBody();
+    } else {
+      return const ConnectedMemoryBody();
```

Comment:
This is fine for this PR, but I think we should try to reuse a single widget. In an ideal world, the MemoryBody widget would be fed data from the memory controller, and it should not need to have many checks for whether we are online or off.

Reply:
Most likely yes.